diff --git a/homeassistant/components/zha/number.py b/homeassistant/components/zha/number.py
index 24964d7a15..4c43958f41 100644
--- a/homeassistant/components/zha/number.py
+++ b/homeassistant/components/zha/number.py
@@ -956,8 +956,8 @@ class ThermostatLocalTempCalibration(ZHANumberConfigurationEntity):
     """Local temperature calibration."""
 
     _unique_id_suffix = "local_temperature_calibration"
-    _attr_native_min_value: float = -2.5
-    _attr_native_max_value: float = 2.5
+    _attr_native_min_value: float = -5.0
+    _attr_native_max_value: float = 5.0
     _attr_native_step: float = 0.1
     _attr_multiplier: float = 0.1
     _attribute_name = "local_temperature_calibration"
